api/src/exam-environment/routes/exam-environment.ts

```diff
@@ -0,0 +1,578 @@
+/* eslint-disable jsdoc/require-returns, jsdoc/require-param */
+import { type FastifyPluginCallbackTypebox } from '@fastify/type-provider-typebox';
+import { type FastifyInstance, type FastifyReply } from 'fastify';
+import jwt from 'jsonwebtoken';
+
+import * as schemas from '../schemas';
+import { mapErr, syncMapErr, UpdateReqType } from '../../utils';
+import { JWT_SECRET } from '../../utils/env';
+import {
+  checkAttemptAgainstGeneratedExam,
+  checkPrerequisites,
+  constructUserExam,
+  userAttemptToDatabaseAttemptQuestionSets,
+  validateAttempt
+} from '../utils/exam';
+import { ERRORS } from '../utils/errors';
+
+/**
+ * Wrapper for endpoints related to the exam environment desktop app.
+ *
+ * Requires exam environment authorization token to be validated.
+ */
+export const examEnvironmentValidatedTokenRoutes: FastifyPluginCallbackTypebox =
+  (fastify, _options, done) => {
+    // TODO: Is there any reason to delete the token without generating a new one?
+    fastify.post(
+      '/exam-environment/exam/generated-exam',
+      {
+        schema: schemas.examEnvironmentPostExamGeneratedExam
+      },
+      postExamGeneratedExamHandler
+    );
+    fastify.post(
+      '/exam-environment/exam/attempt',
+      {
+        schema: schemas.examEnvironmentPostExamAttempt
+      },
+      postExamAttemptHandler
+    );
+    fastify.post(
+      '/exam-environment/screenshot',
+      {
+        schema: schemas.examEnvironmentPostScreenshot
+      },
+      postScreenshotHandler
+    );
+    done();
+  };
+
+/**
+ * Wrapper for endpoints related to the exam environment desktop app.
+ *
+ * Does not require exam environment authorization token to be validated.
+ */
+export const examEnvironmentOpenRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.post(
+    '/exam-environment/token/verify',
+    {
+      schema: schemas.examEnvironmentTokenVerify
+    },
+    tokenVerifyHandler
+  );
+  done();
+};
+
+interface JwtPayload {
+  examEnvironmentAuthorizationToken: string;
+}
+
+/**
+ * Verify an authorization token has been generated for a user.
+ *
+ * Does not require any authentication.
+ *
+ * **Note**: This has no guarantees of which user the token is for. Just that one exists in the database.
+ */
+async function tokenVerifyHandler(
+  this: FastifyInstance,
+  req: UpdateReqType<typeof schemas.examEnvironmentTokenVerify>,
+  reply: FastifyReply
+) {
+  const { 'exam-environment-authorization-token': encodedToken } = req.headers;
+
+  try {
+    jwt.verify(encodedToken, JWT_SECRET);
+  } catch (e) {
+    // TODO: What to send back here? Request is valid, but token is not?
```

Comment:
What's the concern? The main thing is this should send back distinct codes based on what the requester needs to do. If there's only one thing they need to (request a new token), then all failure responses can have the same code.

Reply:
It is not much of a concern, it is more the discussion of how to communicate the fact that a token is not valid through status code:
<details>
  <summary>To use status, or not to use status</summary>

![image](https://github.com/user-attachments/assets/7aaff2d1-9d21-4c0c-a657-10658c043431)

</details>